src/app/campaign/store/actions/campaign-action.creator.ts

```diff
@@ -1,138 +1,90 @@
-  readonly type = ActionTypes.CAMPAIGN_DUP_BY_ID_SUCCESS;
-
-  constructor(public payload: { campaignDoc: HalDoc; flightDocs: HalDoc[]; timestamp?: number }) {
-    this.payload.timestamp = payload.timestamp || Date.now();
-  }
-}
-
-export class CampaignDupByIdFailure implements Action {
-  readonly type = ActionTypes.CAMPAIGN_DUP_BY_ID_FAILURE;
-
-  constructor(public payload: { error: any }) {}
-}
-
-export class CampaignLoad implements Action {
-  readonly type = ActionTypes.CAMPAIGN_LOAD;
-
-  constructor(public payload: { id: number }) {}
-}
-export class CampaignLoadSuccess implements Action {
-  readonly type = ActionTypes.CAMPAIGN_LOAD_SUCCESS;
-
-  constructor(public payload: { campaignDoc: HalDoc; flightDocs: HalDoc[]; flightDaysDocs: { [id: number]: HalDoc[] } }) {}
-}
-export class CampaignLoadFailure implements Action {
-  readonly type = ActionTypes.CAMPAIGN_LOAD_FAILURE;
-
-  constructor(public payload: { error: any }) {}
-}
-
-export class CampaignFormUpdate implements Action {
-  readonly type = ActionTypes.CAMPAIGN_FORM_UPDATE;
-
-  constructor(public payload: { campaign: Campaign; changed: boolean; valid: boolean }) {}
-}
-
-export class CampaignSave implements Action {
-  readonly type = ActionTypes.CAMPAIGN_SAVE;
-
-  constructor(public payload: CampaignFormSave) {}
-}
-export class CampaignSaveSuccess implements Action {
-  readonly type = ActionTypes.CAMPAIGN_SAVE_SUCCESS;
-
-  constructor(
-    public payload: {
-      campaignDoc: HalDoc;
-      deletedFlightDocs: { [id: number]: HalDoc };
-      updatedFlightDocs: { [id: number]: HalDoc };
-      updatedFlightDaysDocs: { [id: number]: HalDoc[] };
-      createdFlightDocs: { [id: number]: HalDoc };
-      createdFlightDaysDocs: { [id: number]: HalDoc[] };
-    }
-  ) {}
-}
-export class CampaignSaveFailure implements Action {
-  readonly type = ActionTypes.CAMPAIGN_SAVE_FAILURE;
-
-  constructor(public payload: { error: any }) {}
-}
-
-export class CampaignDelete implements Action {
-  readonly type = ActionTypes.CAMPAIGN_DELETE;
-
-  constructor(public payload: HalDoc) {}
-}
-export class CampaignDeleteSuccess implements Action {
-  readonly type = ActionTypes.CAMPAIGN_DELETE_SUCCESS;
-
-  constructor(public payload: { id: number | string }) {}
-}
-export class CampaignDeleteFailure implements Action {
-  readonly type = ActionTypes.CAMPAIGN_DELETE_FAILURE;
-
-  constructor(public payload: { error: any }) {}
-}
-
-export class CampaignAddFlight implements Action {
-  readonly type = ActionTypes.CAMPAIGN_ADD_FLIGHT;
-
-  public payload: { campaignId: number | string; flightId: number; startAt: Moment; endAt: Moment };
-
-  constructor(payload: { campaignId: number | string; flightId?: number; startAt?: Moment; endAt?: Moment }) {
+export const CampaignNew = createAction(ActionTypes.CAMPAIGN_NEW);
+
+export const CampaignDuplicate = createAction(ActionTypes.CAMPAIGN_DUPLICATE, props<{ campaign: Campaign; flights: Flight[] }>());
+
```

Comment:
createAction can take `props` or it can take a function that returns props as it's second parameter. props reduces boilerplate.